### DISCUSSION TITLE
Recommended caching behavior when proxying script

### DISCUSSION THREAD

**Discussion by vstollen** [plausible/analytics#1408] [Anything Else]
Currently, I am testing out to proxy Plausible through Firebase Hosting with Firebase Cloud Functions.

This means that all requests are sent to the Firebase / Google CDN and handed to a Cloud Function, if the response is not cached. Do you have a recommended caching behavior? The `cache-control` header sent by Plausible does not allow for caching:
```
cache-control: max-age=0, private, must-revalidate
```

But caching would certainly improve performance and reduce costs by having less function invocations.

**Comment by ukutaht** [comment]
This header is required by Netlify. You can certainly cache it on your end, no problem. We will soon move to a different CDN and we'll start sending better cache headers. Caching on the client side for up to 24 hours should be OK. You don't want to cache forever since we do release updates to the script regularly.

**Comment by fili** [comment]
FYI I applied 24 hours caching on the reverse proxy setup in Google Cloud Run by changing the `cache-control` header, but only for the `plausible.js` script. Left the `cache-control` for `/api/event` as-is. 

Seems to work fine and allows for caching the `plausible.js` script on the Google Cloud CDN.

**Comment by metmarkosaric** [comment]
Our script is now being cached so there should not be a need for any changes anymore
